atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifierTest.java

```diff
@@ -306,23 +307,37 @@ private EndpointDetails createDetails(String dc, String rack, String host) {
         return details.setRack(rack);
     }
 
-    private ImmutableCassandraVerifierConfig.Builder getVerifierConfigBuilderWithDefaults(EndpointDetails... details)
-            throws TException {
+    private ImmutableCassandraVerifierConfig.Builder getBaseVerifierConfigBuilderWithDefaults() {
         return CassandraVerifierConfig.builder()
                 .clientConfig(clientConfig)
-                .servers(setTopologyAndGetServersConfig(details))
                 .keyspace("test")
                 .replicationFactor(3)
                 .ignoreNodeTopologyChecks(false)
                 .ignoreDatacenterConfigurationChecks(false)
                 .schemaMutationTimeoutMillis(0);
     }
 
-    private CassandraServersConfig setTopologyAndGetServersConfig(EndpointDetails... details) throws TException {
-        when(client.describe_ring(CassandraConstants.SIMPLE_RF_TEST_KEYSPACE))
-                .thenReturn(ImmutableList.of(mockRangeWithDetails(details)));
+    private ImmutableCassandraVerifierConfig.Builder getVerifierConfigBuilderWithDefaults() {
+        return getBaseVerifierConfigBuilderWithDefaults().servers(setTopologyAndGetServersConfig());
+    }
+
+    private ImmutableCassandraVerifierConfig.Builder
+            getVerifierConfigBuilderWithDefaultsAndClientDescribeRingInvocation(EndpointDetails... details)
+                    throws TException {
+        return getBaseVerifierConfigBuilderWithDefaults()
+                .servers(setTopologyAndGetServersConfigWithClientDescribeRingInvocation(details));
+    }
+
+    private CassandraServersConfig setTopologyAndGetServersConfig() {
         return ImmutableDefaultConfig.builder()
                 .addThriftHosts(InetSocketAddress.createUnresolved(HOST_1, 8080))
                 .build();
     }
+
+    private CassandraServersConfig setTopologyAndGetServersConfigWithClientDescribeRingInvocation(
+            EndpointDetails... details) throws TException {
+        when(client.describe_ring(CassandraConstants.SIMPLE_RF_TEST_KEYSPACE))
+                .thenReturn(ImmutableList.of(mockRangeWithDetails(details)));
+        return setTopologyAndGetServersConfig();
+    }
```

Comment:
We could also use boolean as a parameter (boolean prepareClientDescribeRingInvocation) but I chose this way (which is way more explicit). However now we have multiple functions. It may be overkill.

Reply:
Why led you to make such refactor? I don't think it's necessarily bad, but I don't see much stuff simplified (other than just things being more explicit now) and we end up with more lines of code?